modules/nw-bgp-examples.adoc

```diff
@@ -0,0 +1,259 @@
+          toAdvertise:
+            allowed:
+              prefixes:
+              - 10.0.1.0/24
+  raw:
+    rawConfig: |-
+      router bgp 64512
+      address-family ipv4 unicast
+        import vrf blue
+        exit-address-family
+      router bgp 64512 vrf blue
+      address-family ipv4 unicast
+        import vrf default
+        exit-address-family
+  nodeSelector:
+    matchLabels:
+      kubernetes.io/hostname: ovn-worker
+----
+
+The generated `FRRConfiguration` object configures the subnet `10.0.1.0/24`, which belongs to VRF blue, is imported into the default VRF and advertised to the `172.18.0.5` neighbor. Because the `targetVRF` uses the default value, the routes are leaked and advertised in the default VRF. Additionally, routes are imported from the default VRF into the blue VRF.
+
+[id="advertising-pod-ips-from-a-user-defined-network-over-bgp-with-vpn_{context}"]
+== Advertising pod IPs from a user-defined network over BGP with VPN
+
+In this scenario, a VLAN interface attached to a VRF device is associated with the blue network and the VLAN includes the external PE router. This setup provides a _VRF lite_ design where FRR-K8S is leveraged to advertise the blue network over only the corresponding VRF/VLAN link to the next hop PE router. The red tenant uses the same configuration.
+
+Red user-defined network::
+- A VRF named `mp0-red`
+- A VLAN interface attached to the VRF device and connected to the external PE router
+- An assigned subnet of `10.0.2.0/24`
+
+Blue user-defined network::
+- A VRF named `mp0-blue`
+- A VLAN interface attached to the VRF device and connected to the external PE router
+- An assigned subnet of `10.0.1.0/24`
+
+[NOTE]
+====
+This approach is available only when using OVN-Kubernetes local gateway mode by setting `routingViaHost=true`.
+====
+
+In the following configuration, an additional `FRRConfiguration` CR configures peering with the PE router on the blue and red VLANs:
+
+.`FRRConfiguration` CR manually configured for BGP VPN setup
+[source,yaml]
+----
+apiVersion: frrk8s.metallb.io/v1beta1
+kind: FRRConfiguration
+metadata:
+  name: vpn-ovn-worker
+  namespace: openshift-frr-k8s
+  labels:
+    routeAdvertisements: vpn-blue-red
+spec:
+  bgp:
+    routers:
+    - asn: 64512
+      vrf: blue
+      neighbors:
+      - address: 182.18.0.5
+        asn: 64512
+        disableMP: false
+        holdTime: 1m30s
+        keepaliveTime: 30s
+        passwordSecret: {}
+        port: 179
+        toAdvertise:
+          allowed:
+            mode: filtered
+        toReceive:
+          allowed:
+            mode: filtered
+    - asn: 64512
+      vrf: red
+      neighbors:
+      - address: 192.18.0.5
+        asn: 64512
+        disableMP: false
+        holdTime: 1m30s
+        keepaliveTime: 30s
+        passwordSecret: {}
+        port: 179
+        toAdvertise:
+          allowed:
+            mode: filtered
+        toReceive:
+          allowed:
+            mode: filtered
+----
```

Comment:
So this is the FRRConfiguration the generated ones will be based on.

I would probably simplify it a bit:
```
apiVersion: frrk8s.metallb.io/v1beta1
kind: FRRConfiguration
metadata:
  name: vpn-blue-red
  namespace: openshift-frr-k8s
  labels:
    routeAdvertisements: vpn-blue-red
spec:
  bgp:
    routers:
    - asn: 64512
      vrf: blue
      neighbors:
      - address: 182.18.0.5
        asn: 64512
        toReceive:
          allowed:
            mode: filtered
    - asn: 64512
      vrf: red
      neighbors:
      - address: 192.18.0.5
        asn: 64512
        toReceive:
          allowed:
            mode: filtered
```